File: app/src/main/java/com/hydrocare/app/MainActivity.java
package com.hydrocare.app;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.ProgressBar;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import com.google.android.material.button.MaterialButton;
import java.time.LocalDate;

public class MainActivity extends AppCompatActivity {

    private static final int GOAL = 2000;
    private SharedPreferences prefs;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        prefs = getSharedPreferences("hydrocare", MODE_PRIVATE);
        resetIfNewDay();

        TextView tvIntake = findViewById(R.id.tvIntake);
        TextView tvGoal = findViewById(R.id.tvGoalLabel);
        TextView tvMotivation = findViewById(R.id.tvMotivation);
        ProgressBar progressBar = findViewById(R.id.progressBar);

        MaterialButton btn250 = findViewById(R.id.btn250);
        MaterialButton btn500 = findViewById(R.id.btn500);
        MaterialButton btn750 = findViewById(R.id.btn750);
        MaterialButton btnReset = findViewById(R.id.btnReset);

        Runnable refresh = () -> {
            int intake = prefs.getInt("intake", 0);
            tvIntake.setText(String.valueOf(intake));
            tvGoal.setText(intake + " / " + GOAL + " ml");
            progressBar.setProgress(Math.min(intake, GOAL));
            if (intake >= GOAL) {
                tvMotivation.setText("You crushed your goal today!");
            } else if (intake >= GOAL / 2) {
                tvMotivation.setText("Halfway there, keep it up!");
            } else if (intake > 0) {
                tvMotivation.setText("Good start, keep drinking!");
            } else {
                tvMotivation.setText("Tap a button below to log your first drink!");
            }
        };

        btn250.setOnClickListener(v -> { add(250); refresh.run(); });
        btn500.setOnClickListener(v -> { add(500); refresh.run(); });
        btn750.setOnClickListener(v -> { add(750); refresh.run(); });
        btnReset.setOnClickListener(v -> {
            prefs.edit().putInt("intake", 0).apply();
            refresh.run();
        });

        refresh.run();
    }

    private void add(int ml) {
        int current = prefs.getInt("intake", 0);
        prefs.edit().putInt("intake", current + ml).apply();
    }

    private void resetIfNewDay() {
        String today = LocalDate.now().toString();
        if (!today.equals(prefs.getString("day", ""))) {
            prefs.edit().putInt("intake", 0).putString("day", today).apply();
        }
    }
}

File: app/src/main/java/com/hydrocare/app/MainActivity.kt
package com.hydrocare.app

import android.os.Bundle
import android.widget.ProgressBar
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.button.MaterialButton
import java.time.LocalDate

class MainActivity : AppCompatActivity() {

    private val prefs by lazy { getSharedPreferences("hydrocare", MODE_PRIVATE) }
    private val goal = 2000

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        resetIfNewDay()

        val tvIntake = findViewById<TextView>(R.id.tvIntake)
        val tvGoal = findViewById<TextView>(R.id.tvGoalLabel)
        val tvMotivation = findViewById<TextView>(R.id.tvMotivation)
        val progressBar = findViewById<ProgressBar>(R.id.progressBar)

        fun updateUI() {
            val intake = prefs.getInt("intake", 0)
            tvIntake.text = intake.toString()
            tvGoal.text = "$intake / $goal ml daily goal"
            progressBar.progress = minOf(intake, goal)
            tvMotivation.text = when {
                intake >= goal -> "Amazing! You've hit your goal today!"
                intake >= goal / 2 -> "Keep going, you're doing great!"
                intake > 0 -> "Good start, keep drinking!"
                else -> "Start your hydration journey!"
            }
        }

        fun addWater(ml: Int) {
            val current = prefs.getInt("intake", 0)
            prefs.edit().putInt("intake", current + ml).apply()
            updateUI()
        }

        findViewById<MaterialButton>(R.id.btn250).setOnClickListener { addWater(250) }
        findViewById<MaterialButton>(R.id.btn500).setOnClickListener { addWater(500) }
        findViewById<MaterialButton>(R.id.btn750).setOnClickListener { addWater(750) }
        findViewById<MaterialButton>(R.id.btnReset).setOnClickListener {
            prefs.edit().putInt("intake", 0).apply()
            updateUI()
        }

        updateUI()
    }

    private fun resetIfNewDay() {
        val today = LocalDate.now().toString()
        val lastDay = prefs.getString("last_day", "")
        if (lastDay != today) {
            prefs.edit().putInt("intake", 0).putString("last_day", today).apply()
        }
    }
}
